param/__init__.py

```diff
@@ -1359,10 +1359,11 @@ class List(Parameter):
     items in the list are checked to be of that type.
     """
 
-    __slots__ = ['class_','bounds']
+    __slots__ = ['bounds', 'item_type', 'class_']
 
-    def __init__(self,default=[],class_=None,instantiate=True,
+    def __init__(self,default=[],class_=None,item_type=None,instantiate=True,
                  bounds=(0,None),**params):
+        self.item_type = item_type or class_
         self.class_ = class_
```

Comment:
For backward compatibility this should also reflect the `item_type`. Is it possible to simply make `class_` a property?

Reply:
Just as an example, this change will already break the serialization code if a user uses item_type rather than `class_`.

Reply:
> Is it possible to simply make `class_` a property?

I don't think so...I hit an issue when importing holoviews where `class_` as a slot was expected which is why I re-added it. What I should do is keep them in sync, so I'll push this change:

```python
self.class_ = item_type or class_
```

Which should fix the serialization issue.

Reply:
Addressed in 9096e39

Reply:
Won't there still be an issue when deserializing between different param versions, such that class_ was serialized previously but gets deserialized only to class_ and not iterm_type?  Seems like there needs to be the same sort of logic in get_state.

Reply:
Annoying but yes, good point!